app/controllers/comments_controller.rb

```diff
@@ -19,6 +19,7 @@ class CommentsController < ApplicationController
   before_action :check_permission_to_edit, only: [:edit, :update ]
   before_action :check_permission_to_delete, only: [:delete_comment, :destroy]
   before_action :check_parent_comment_permissions, only: [:new, :create, :add_comment_reply]
+  before_action :check_guest_comment_permissions, only: [:new, :create, :add_comment_reply]
```

Comment:
Could we change this to something like `check_site_comment_permissions` or `check_comment_admin_setting` for clarity?